### PR TITLE
Use Gradle Catalog to manage lib versions.

### PR DESCRIPTION
**Pull request by dev-weiqi**
see more details in [Sharing dependency versions between projects](https://docs.gradle.org/current/userguide/platforms.html )

### CODE REVIEW

**Review by AmyPlace — APPROVED** [review]
LGTM

**Review by ivanisidrowu — APPROVED** [review]
Good job!

**Review by lush-life — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/java/Version.kt [hunk 1]
```diff
@@ -14,7 +14,12 @@
  *    limitations under the License.
  */
 
-object AndroidVersions {
+object Version {
+    const val applicationId = "tw.ktrssreader"
+    const val compileSdk = 31
```
[the remark this answers is not in the document]
Comment (dev-weiqi) [reply buildSrc/src/main/java/Version.kt:19]: Also upgrade target sdk version to 31!

File: android/build.gradle.kts [hunk 2]
```diff
@@ -34,39 +31,49 @@ android {
...
+    androidTestImplementation(libs.mockk)
+    androidTestImplementation(libs.junitAndroid)
+    androidTestImplementation(libs.espressoCore)
+}
+
+publishing {
+    publications {
+        register<MavenPublication>("release") {
+            groupId = "com.github.ivanisidrowu"
+            artifactId = "KtRssReader"
```
[the remark this answers is not in the document]
Comment (dev-weiqi) [reply android/build.gradle.kts:71]: Our artifact id is uppercase now 🤣

截圖 2022-07-29 10 07 11

File: buildSrc/src/main/java/Version.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (lush-life) [reply buildSrc/src/main/java/Version.kt:19]: [Suggestion] Upgrade major version of buildTool that equals to compileSdk version

[the remark this answers is not in the document]
Comment (dev-weiqi) [reply buildSrc/src/main/java/Version.kt:19]: Done!

**Review by lush-life — APPROVED** [review]

File: buildSrc/src/main/java/Version.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (lush-life) [reply buildSrc/src/main/java/Version.kt:19]: 👍

**Pull request merged**